### ISSUE TITLE
If the delay or delayOpen value is set, the whisper overlay will not open. 

### ISSUE THREAD

**Issue by jngyunkim** [rsuite/rsuite#3490]
### What version of rsuite are you using?

5.46.1

### What version of React are you using?

18.0

### What version of TypeScript are you using (if any)?

4.9.4

### Describe the Bug

If the delay or delayOpen property is set, the whisper overlay will not open. 

### Expected Behavior

My understanding is that if delay is set, the overlay will open and close after delay. 

### To Reproduce

https://rsuitejs.com/components/whisper/
In the example in the Whisper docs, setting delay makes the bug.

```ts
import { Whisper, Button } from 'rsuite';

const Overlay = React.forwardRef(({ style, onClose, ...rest }, ref) => {
  const styles = {
    ...style,
    color: '#000',
    background: '#fff',
    width: 200,
    padding: 10,
    borderRadius: 4,
    position: 'absolute',
    border: '1px solid #ddd',
    boxShadow: '0 3px 6px -2px rgba(0, 0, 0, 0.6)'
  };

  return (
    <div {...rest} style={styles} ref={ref}>
      Overlay
      <hr />
      <button onClick={onClose}>close</button>
    </div>
  );
});

const App = () => (
  <Whisper
    trigger="click"
    delay={10}
    speaker={(props, ref) => {
      const { className, left, top, onClose } = props;
      return <Overlay style={{ left, top }} onClose={onClose} className={className} ref={ref} />;
    }}
  >
    <Button>Open</Button>
  </Whisper>
);

ReactDOM.render(<App />, document.getElementById('root'));
```

**Comment by simonguo** [comment]
Fixed in v5.48.0

**Comment by jngyunkim** [comment]
Thanks a lot for fixing this!
